resources/views/presences/index.blade.php

```diff
@@ -101,41 +69,91 @@ class="btn btn-success btn-round" data-original-title="" title="marquer les pré
+                    { data: 'concernes', name: 'concernes' },
+                    { data: 'date_debut', name: 'date_debut' },
+                    { data: 'date_fin', name: 'date_fin' },
+                    { data: 'heure_debut', name: 'heure_debut' },
+                    { data: 'actions', name: 'actions' },
+                ],
+                paging: true,
                 layout: {
                     topStart: {
                         buttons: [
                             {
-                                title: null,
-                                extend: 'csv',
+                                text: '<i class="material-icons">file_download</i> Exporter Excel',
+                                className: 'btn btn-success btn-round',
                                 filename: $fileName,
+                                action: function () {
+                                    window.location.href = "{{ route('presences.activites.export') }}";
+                                }
                             },
-                            {
+                            /*{
                                 title: null,
                                 extend: 'excel',
-                                filename: $fileName
-                            },
+                                filename: $fileName,
+                                exportOptions: {
+                                    columns: ':not(:last-child)',
+                                }
+                            },*/
                             {
                                 title: null,
                                 extend: 'print',
-                                filename: $fileName
+                                filename: $fileName,
+                                exportOptions: {
+                                    columns: ':not(:last-child)',
+                                }
                             }
                         ]
                     }
                 },
                 "pagingType": "full_numbers",
                 "lengthMenu": [
-                    [10, 25, 50, -1],
-                    [10, 25, 50, "All"]
+                    [50, 100, 150, -1],
+                    [50, 100, 150, "All"]
                 ],
-                "order": [],
+                "order": [[1, "asc"]],
                 responsive: true,
-                language: datatable_fr
+                language: datatable_fr,
+                initComplete: function () {
+                    // Apply the search
+                    this.api()
+                            .columns()
+                            .every(function () {
+                                var that = this;
+
+                                $('input', this.header()).on('keyup change clear', function () {
+                                    if (that.search() !== this.value) {
+                                        console.log('searching...', this.value);
```

Comment:
out of curiousity is this console logging synchronous? It could be very inefficient if some not a big deal though.